File: backup_antes_historico/backend_app/schemas/ticket.py
from datetime import datetime

from pydantic import BaseModel, Field

from app.models.enums import TicketPriority, TicketStatus
from app.schemas.user import UserPublic


class TicketCreate(BaseModel):
    title: str = Field(min_length=3, max_length=180)
    description: str = Field(min_length=5)
    category: str = Field(min_length=2, max_length=60)
    priority: TicketPriority = TicketPriority.medium


class TicketUpdate(BaseModel):
    status: TicketStatus | None = None
    priority: TicketPriority | None = None
    assignee_id: int | None = None


class TicketOut(BaseModel):
    id: int
    title: str
    description: str
    category: str
    priority: TicketPriority
    status: TicketStatus
    creator_id: int
    assignee_id: int | None
    created_at: datetime
    updated_at: datetime
    creator: UserPublic | None = None
    assignee: UserPublic | None = None

    model_config = {"from_attributes": True}


class TicketListOut(BaseModel):
    items: list[TicketOut]
    total: int
    skip: int
    limit: int


class CommentCreate(BaseModel):
    content: str = Field(min_length=1, max_length=4000)


class CommentOut(BaseModel):
    id: int
    ticket_id: int
    author_id: int
    content: str
    created_at: datetime
    author: UserPublic | None = None

    model_config = {"from_attributes": True}

File: flutter_src/lib/models/ticket.dart
import 'user.dart';

class Ticket {
  final int id;
  final String title;
  final String description;
  final String category;
  final String priority;
  final String status;
  final int creatorId;
  final int? assigneeId;
  final DateTime createdAt;
  final DateTime updatedAt;
  final User? creator;
  final User? assignee;

  const Ticket({
    required this.id,
    required this.title,
    required this.description,
    required this.category,
    required this.priority,
    required this.status,
    required this.creatorId,
    required this.assigneeId,
    required this.createdAt,
    required this.updatedAt,
    this.creator,
    this.assignee,
  });

  factory Ticket.fromJson(Map<String, dynamic> json) => Ticket(
        id: json['id'],
        title: json['title'],
        description: json['description'],
        category: json['category'],
        priority: json['priority'],
        status: json['status'],
        creatorId: json['creator_id'],
        assigneeId: json['assignee_id'],
        createdAt: DateTime.parse(json['created_at']),
        updatedAt: DateTime.parse(json['updated_at']),
        creator: json['creator'] != null ? User.fromJson(json['creator']) : null,
        assignee: json['assignee'] != null ? User.fromJson(json['assignee']) : null,
      );
}

class TicketPage {
  final List<Ticket> items;
  final int total;
  final int skip;
  final int limit;

  const TicketPage({required this.items, required this.total, required this.skip, required this.limit});

  factory TicketPage.fromJson(Map<String, dynamic> json) => TicketPage(
        items: (json['items'] as List).map((e) => Ticket.fromJson(e)).toList(),
        total: json['total'],
        skip: json['skip'],
        limit: json['limit'],
      );
}
